### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by raphaelrobert**
This PR fixes the clippy warnings & introduces `rustfmt.toml`.

Note that `wrap_comments` is currently not stable, so once in a while nightly needs to do a pass on the code base.

### CODE REVIEW

File: rustfmt.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+comment_width = 80
+wrap_comments = true
+edition = "2018"
+format_code_in_doc_comments = true
```
Comment (franziskuskiefer) [inline rustfmt.toml:4]: newline

File: rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+comment_width = 80
+wrap_comments = true
```
Comment (franziskuskiefer) [inline rustfmt.toml:2]: Hm so it's a nightly feature... This produces a lot of not so nice warnings.
But I'm actually more concerned about stability. How stable is this feature, i.e. how much churn does it produce?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Generally looks good. But one question for the `wrap_comments`.

While you're at it can you also fix all the warnings in the `managed_group`?

File: rustfmt.toml (hunk 2 above)
Comment (raphaelrobert) [reply rustfmt.toml:2]: It seems to be quite stable, the only problem that was reported was when you use markup tables in comments. I guess for now we have to run it manually once in a while to clean up comments.

**Comment by raphaelrobert** [comment]
> While you're at it can you also fix all the warnings in the `managed_group`?

That's already fixed.

**Review by franziskuskiefer — APPROVED** [review]
> > While you're at it can you also fix all the warnings in the `managed_group`?
> 
> That's already fixed.

Ah totally missed that, sorry 🙇 

Ok let's do this.

**Pull request merged**